cmd/query/app/querysvc/metrics_query_service.go

```diff
@@ -16,22 +16,18 @@ package querysvc
 
 import (
 	"context"
-	"errors"
 	"time"
 
 	"github.com/jaegertracing/jaeger/proto-gen/api_v2/metrics"
 	"github.com/jaegertracing/jaeger/storage/metricsstore"
 )
 
-// MetricsQueryService contains the underlying reader required for querying the metrics store.
+// MetricsQueryService provides a means of querying R.E.D metrics from an underlying metrics store.
 type MetricsQueryService struct {
```

Comment:
What is `MetricsQueryService` supposed to add on top of `metricsstore.Reader`?

Note that we used to have HTTP handler use SpanReader directly, but then refactored it into QueryService because we needed to share it with GRPC Handler **and** the query service was an abstraction on top of two span readers: primary and archive. In your case there doesn't seem to be any additional abstraction needed, you just proxy all calls into the reader directly. Not a huge issue to leave as is, but just curious if there's any thinking that a layer of abstraction would be needed in the future.

Reply:
Indeed `MetricsQueryService` adds little on top of `metricsstore.Reader`. It has exactly the same function signature as the Reader and yes, it simply forwards the call. I also don't see a need for multiple metrics readers at runtime, for now at least.

It was added to follow the same design as span/dependency querying to avoid confusing/surprising those following the code and seeing a `QueryService` being used by handlers to query spans/dependencies, then a `Reader` used to query metrics; when they may expect the handlers to refer to "QueryService" abstractions for querying spans, dependencies **and** metrics.

One possibility is to avoid confusion, yet minimizing unnecessary code, is to make `MetricsQueryService` an interface, embedding `metricsstore.Reader`, meaning we can remove the `MetricsQueryService` struct and its proxy functions altogether and simply give the handlers a `metricsstore.Reader` instance. That way, handlers still refer to the "MetricsQueryService" alias for metrics, but instead using a Reader implementation. Something like:
```
type MetricsQueryService interface {
	metricsstore.Reader
}
```